Tasks/AzureAppServiceManage/azureappservicemanage.ts

```diff
@@ -0,0 +1,97 @@
+        var preserveVnet: boolean = tl.getBoolInput('PreserveVnet', false);
+        var endPointAuthCreds = tl.getEndpointAuthorization(connectedServiceName, true);
+        var subscriptionId = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+        var taskResult = true;
+
+        var endPoint = new Array();
+        endPoint["servicePrincipalClientID"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'serviceprincipalid', true);
+        endPoint["servicePrincipalKey"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'serviceprincipalkey', true);
+        endPoint["tenantID"] = tl.getEndpointAuthorizationParameter(connectedServiceName, 'tenantid', true);
+        endPoint["subscriptionId"] = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true);
+        endPoint["url"] = tl.getEndpointUrl(connectedServiceName, true);
+
+        if(resourceGroupName === null) {
+            resourceGroupName = await azureRmUtil.getResourceGroupName(endPoint, webAppName);
+        }
+        switch(action) {
+            case "Start Azure App Service": {
+                tl._writeLine(await azureRmUtil.startAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Stop Azure App Service": {
+                tl._writeLine(await azureRmUtil.stopAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Restart Azure App Service": {
+                tl._writeLine(await azureRmUtil.restartAppService(endPoint, resourceGroupName, webAppName));
+                break;
+            }
+            case "Swap Slots": {
+                await swapSlot(endPoint, resourceGroupName, webAppName, sourceSlot, swapWithProduction, targetSlot, preserveVnet);
+                break;
+            }
+            default:
+                throw Error(tl.loc('InvalidAction'));
+        }
+    }
+    catch(exception)
+    {
+        taskResult = false;
+        tl.setResult(tl.TaskResult.Failed, exception);
+    }
+    var customMessage = {
+        type: action
+    }
+    var deploymentId = kuduLogUtil.generateDeploymentId();
+
+    if(action === "Swap Slots") {
+        customMessage['type'] = 'SlotSwap'; // for Ibiza CD flow
+        customMessage['sourceSlot'] = sourceSlot;
+        customMessage['targetSlot'] = swapWithProduction ? "Production" : targetSlot;
+
+        await updateKuduDeploymentLog(endPoint, webAppName, resourceGroupName, true, sourceSlot, taskResult, customMessage, deploymentId);
+        await updateKuduDeploymentLog(endPoint, webAppName, resourceGroupName, !(swapWithProduction), targetSlot, taskResult, customMessage, deploymentId);
+    }
+    else {
+        customMessage['slotName'] = 'Production';
+        await updateKuduDeploymentLog(endPoint, webAppName, resourceGroupName, false, null, taskResult, customMessage, deploymentId);
```

Comment:
what is the purpose of pushing logs to kudu for other azure app service manage actions ?